chore: clean up deps and upgrade node version

crondinini opened the pull request:
### Overview 

This PR does the following:

- removes a few dev deps, namely
   - css related dependencies that seemed to have been previously used in storybook
   - everything related to publishing and semantic release (instead of that, I created [an issue](https://github.com/Developer-DAO/web3-ui/issues/18) for us to look into that)
      - this included removing an action from Github action 
   - commit lint for conventional commits 
- upgrades the node version used
- reverts adding everything to the stage during the pre-commit hook
- deleted the changelog that was being automatically generated

### Excluded from the PR

There are more things we could look into simplifying that I didn't include here due to my own time constraints:
- Both rollup and webpack are being referenced. It seems that storybook requires webpack. We could look into using only one bundler that works for both things. I've created [an issue](https://github.com/Developer-DAO/web3-ui/issues/19) for that.
- Initially I deleted the whole set up around the templates with hygen, but then I saw commits that indicated to me they were being used.
- I also didn't deleted the `Examples.tsx` but it seems that we could. Why do we need both that and the templates? But I didn't want to create conflicts in other PRs so I left it as is.

Nazeeh21 commented on .husky/pre-commit:
```diff
@@ -2,4 +2,3 @@
 . "$(dirname "$0")/_/husky.sh"
 
 yarn format
-git add .
```
Just curiously asking is there any particular reason for removing this?

Nazeeh21 replied:
Overall changes LGTM, though would love to hear from @Dhaiwat10.

crondinini replied:
@Nazeeh21 This was adding every file change made to the commit, even when I was selecting just some to be staged.

Nazeeh21 commented:
Overall changes LGTM, though would love to hear from @Dhaiwat10.

with-heart commented on package.json:
```diff
@@ -25,8 +29,8 @@
-    "publish": "npx semantic-release --no-ci",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "publish": "npm publish"
```
We should remove this line because it's redundant!

crondinini replied:
Yes, that's true! Will remove.

Re: Husky, I agree with using lint-staged with husky, that's how I've used husky in the past. Do you think it's worth doing it on this PR?

with-heart replied:
Hmmmm. Actually now that you mention it, if you're down for it, wanna create a separate PR? It can serve as a shining example of what the transition from `husky` standalone to `husky`+`lint-staged` looks like.

crondinini replied:
Sure! I can def do that :)

with-heart reviewed (approved):
Love the intent behind this PR and it's a good clean up!

I think we'd get more out of using `husky`+`lint-staged` so we can define our pre-commit tasks directly in `package.json`. Having them hidden in `.husky/pre-commit` isn't great imo.

This is great though!

Dhaiwat10 commented:
@crondinini you can safely remove the `Example` component/folder. It's of no use to us now.

Dhaiwat10 commented on .nvmrc:
```diff
@@ -1 +1 @@
-12
+16.13.0
```
👌

Dhaiwat10 commented on package.json:
```diff
@@ -62,12 +59,10 @@
     "babel-loader": "^8.2.1",
     "classnames": "^2.2.6",
-    "conventional-changelog-conventionalcommits": "^4.4.0",
-    "css-loader": "^5.0.1",
```
Sure this won't break loading `.css` files?

crondinini replied:
Everything builds and storybook runs so for now yes. I never had to use a css loader in the past for loading CSS, I'm guessing it was always handled by the bundler, so maybe it could be done as part of the bundler config issue for example.

Dhaiwat10 reviewed (approved):
Thank you!

crondinini commented:
I've addressed all comments so I'll go ahead and merge this. :) 

Btw, I've enabled actions to run on pull requests so we can avoid getting to a bad merge stage.
